### PR TITLE
Protect API with Fail2ban

### PR DESCRIPTION
**Pull request by lapineige**
## Problem
- API is not protected by fail2ban and could be used for password brute-force. #62

## Solution
- Add fail2ban config

## PR Status
- [ ] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

### CODE REVIEW

**Comment by lapineige** [comment]
!testme

**Comment by lapineige** [comment]
Does that CI thing work with a draft pull request ? 🤔

**Comment by lapineige** [comment]
!testme

File: scripts/_common.sh [hunk 1]
```diff
@@ -25,6 +25,70 @@ function set_permissions {
...
+#=================================================
+
+# Create a dedicated fail2ban config (jail and filter conf files)
+#
+# usage: ynh_add_fail2ban_config log_file filter [max_retry [ports]]
+# | arg: log_file - Log file to be checked by fail2ban
+# | arg: failregex - Failregex to be looked for by fail2ban
+# | arg: max_retry - Maximum number of retries allowed before banning IP address - default: 3
+# | arg: ports - Ports blocked for a banned IP address - default: http,https
+ynh_add_fail2ban_config_temp () {
```
Comment (plopoyop) [inline scripts/_common.sh:39]: the CI says : 
```
ynh_remove_fail2ban_config is now an official helper since version '3.5.0'
```
You may remove this part

**Review by plopoyop — CHANGES_REQUESTED** [review]
there is this warning in your branch which is not in the master / testing branch :
```
41468 WARNING Too many arguments ! "1" will be ignored.
41469 INFO [###############+....] > Adding fail2ban configuration...
41804 WARNING Too many arguments ! "1" will be ignored.
```

File: scripts/_common.sh (hunk 1 above)
Comment (lapineige) [reply scripts/_common.sh:39]: I didn't know that we had an official helper for that…
I'll change it, but I will probably need to try it as I don't know how this new one works…

**Comment by lapineige** [comment]
For the record, I don't have time to work on this now… if anyone is willing to give me an hand… :)

**Comment by lapineige** [comment]
I did not have time to work on this, any help in updating it is appreciated ! 🙂

**Comment by Alkarex** [comment]
Let me know whether anything should be added at FreshRSS level to facilitate the detection

**Comment by lapineige** [comment]
Thanks ! :)

For now I don't think it's needed, we have the regex to find in the log, it's just a matter of converting our way of configuring it to Yunohost helper https://yunohost.org/fr/packaging_apps_helpers#fail2ban (in a word : simplifying the matter by a lot), and cleaning the code next. And testing it.
Not much work, but I'm not available for it, just that was a little :up: message just in case someone wants to try it :)

**Comment by Alkarex** [comment]
See also https://github.com/FreshRSS/FreshRSS/issues/3224
And a recent rework in FreshRSS https://github.com/FreshRSS/FreshRSS/pull/5549
Feedback welcome

**Comment by OniriCorpe** [comment]
hi!

I've done the job here: #171

sorry I can't commit to that draft, so I had to open a new PR 😕

**Pull request closed without merging**